runtime/src/accounts_db.rs

```diff
@@ -2807,58 +2839,125 @@ impl AccountsDB {
 
     // Force flush the cached roots, flush any unrooted frozen slots as well if there are
     // > MAX_CACHE_SLOTS of them.
-    pub fn force_flush_accounts_cache(&self) {
-        self.flush_accounts_cache(true);
+    pub fn force_flush_accounts_cache(&self, max_clean_root: Option<Slot>) {
+        self.flush_accounts_cache(true, max_clean_root);
     }
 
-    pub fn flush_accounts_cache_if_needed(&self) {
-        self.flush_accounts_cache(false);
+    pub fn flush_accounts_cache_if_needed(&self, max_clean_root: Option<Slot>) {
+        self.flush_accounts_cache(false, max_clean_root);
     }
 
-    fn flush_accounts_cache(&self, force_flush: bool) {
-        if !force_flush && self.accounts_cache.num_slots() <= MAX_CACHE_SLOTS {
-            return;
-        }
+    fn flush_rooted_slots_cache(
```

Comment:
nits: `flush_rooted_accounts_cache`? (The body looks a helper fn to `flush_accounts_cache`, not to `flush_slot_cache`).

Also, how about moving this fn beflow `flush_accounts_cache` to minimize the diff?